MB-54166 Fix build for 'backup' in CV caused by missing 'cbauth' dependency

jamesl33 opened the pull request:
As part of the work for MB-54166, the `backup` repository switched from a standalone build to one that requires local dependencies (cbauth).

To fix this:
1. The clone stages for `backup` and `tools-common` were separated.
2. `backup` is now clone used `repo` (with the `backup`) group.
3. The separate `CURRENT_PROJECT` and `CB_SERVER_SOURCE_PROJECT` variables have been merged (required as `backup` and `tools-common` must be in a common location to run linting et al.

This ultimately means that `backup` is cloned using `repo` and built with the expected dependencies (`cbauth`) where it wasn't previously. 

Worth noting that this change has the side-effect of cloning `tools-common` into the root of the Couchbase Server install which is fine as it will be ignored and there's no clashing project names.

jamesl33 committed:
Separate out the clone stage for 'backup' and 'tools-common'

rdemellow commented on Jenkinsfile:
```diff
@@ -33,13 +33,11 @@ pipeline {
         CB_SERVER_SOURCE = "${WORKSPACE}/server"
         CB_SERVER_SOURCE_PROJECT = "${CB_SERVER_SOURCE}/${GERRIT_PROJECT}"
 
+        EXAMINADOR = "${WORKSPACE}/examinador"
```
Can we separate out concerns in this patch, this is a tidy up rather than a required change. If you want to keep it add a comment in the commit message

jamesl33 replied:
Will revert this change, please let me know if I miss any others.

I'll leave the stuff that removed duplication between backup/tools-common though for now.

rdemellow commented on Jenkinsfile:
```diff
@@ -120,56 +118,92 @@ pipeline {
+        stage("Clone") {
+            when {
+                expression {
+                    return env.GERRIT_PROJECT != 'tools-common';
```
Maybe add a comment to why != tool-common? Who's the special case?

rdemellow commented on Jenkinsfile:
```diff
@@ -120,56 +118,92 @@ pipeline {
+                // Initialize and sync 'backup' group using 'repo'
+                dir("${CB_SERVER_SOURCE}") {
+                    sh "repo init -u https://github.com/couchbase/manifest -m ${CB_SERVER_MANIFEST} -g backup"
+                    sh "repo sync --jobs=8"
```
Please use the PARALLELISM env for the number jobs

jamesl33 replied:
Good catch, didn't know about that :+1:

rdemellow commented on Jenkinsfile:
```diff
@@ -120,56 +118,92 @@ pipeline {
-                    }
+                // Initialize and sync 'backup' group using 'repo'
+                dir("${CB_SERVER_SOURCE}") {
+                    sh "repo init -u https://github.com/couchbase/manifest -m ${CB_SERVER_MANIFEST} -g backup"
```
Could we make use backup as an env so that we can just change that in the future?

rdemellow commented:
Can you git the PR a better title and add more information to the commit message :)

jamesl33 committed:
MB-54166 Revert moving the 'EXAMINADOR' variable

jamesl33 committed:
MB-54166 Comment why stages are conditional

jamesl33 committed:
MB-54166 Add a 'CB_SERVER_MANIFEST_GROUPS' variable

This may be used to specify the groups to clone when initializing the
Couchbase Server source directory.

jamesl33 committed:
MB-54166 Use 'env.PARALLELISM' instead of hardcoding jobs

jamesl33 commented:
> Can you git the PR a better title and add more information to the commit message :)

Done :+1:
